python/ccf/ledger.py

```diff
@@ -1001,8 +1001,19 @@ def get_latest_public_state(self) -> Tuple[dict, int]:
         latest_seqno = 0
         for chunk in self:
             for tx in chunk:
-                latest_seqno = tx.get_public_domain().get_seqno()
-                for table_name, records in tx.get_public_domain().get_tables().items():
+                # If a transaction cannot be read (e.g. because it was only partially written to disk
+                # before a crash), return public state so far. This is consistent with CCF's behaviour
+                # which discards the incomplete transaction on recovery.
+                try:
+                    public_domain = tx.get_public_domain()
+                except Exception:
+                    print(
+                        f"Error reading ledger entry. Latest read seqno: {latest_seqno}"
+                    )
+                    return public_tables, latest_seqno
```

Comment:
Also adding some resilience to the Python infra ledger reading that sometimes fails in the new `recovery_benchmark.sh`: https://dev.azure.com/MSRC-CCF/CCF/_build/results?buildId=45908&view=logs&j=5435e0ac-25e5-5426-50be-61b0d0ea8d34&t=20753e5b-87ba-5bdc-0f17-d34c09f927a7&l=195